### PR TITLE
osu.trans fallback support

### PR DESCRIPTION
**Pull request by Nekonyx**
image
to
image

### CODE REVIEW

**Comment by nanaya** [comment]
What does this fix?

**Comment by kj415j45** [comment]
@nanaya profile page die when you're using locale(except en), it fix that.

File: resources/assets/coffee/osu_common.coffee [hunk 1]
```diff
@@ -216,11 +216,14 @@
...
   trans: (key, replacements) ->
-    message = Lang.get key, replacements, currentLocale
-
-    if message == key
-      message = Lang.get key, replacements, fallbackLocale
-
+    message = undefined
+    try
+      message = Lang.get key, replacements
+      message == key and new Error 'need fallback'
```
Comment (nanaya) [inline resources/assets/coffee/osu_common.coffee]: not needed

File: resources/assets/coffee/osu_common.coffee [hunk 2]
```diff
@@ -216,11 +216,14 @@
...
 
   trans: (key, replacements) ->
-    message = Lang.get key, replacements, currentLocale
-
-    if message == key
-      message = Lang.get key, replacements, fallbackLocale
-
+    message = undefined
+    try
+      message = Lang.get key, replacements
```
Comment (nanaya) [inline resources/assets/coffee/osu_common.coffee]: skip the assignment (so it immediately returns)

File: resources/assets/coffee/osu_common.coffee [hunk 3]
```diff
@@ -216,11 +216,14 @@
 
 
   trans: (key, replacements) ->
-    message = Lang.get key, replacements, currentLocale
-
-    if message == key
-      message = Lang.get key, replacements, fallbackLocale
-
+    message = undefined
```
Comment (nanaya) [inline resources/assets/coffee/osu_common.coffee]: not needed

File: resources/assets/coffee/osu_common.coffee [hunk 4]
```diff
@@ -216,11 +216,14 @@
...
+    try
+      message = Lang.get key, replacements
+      message == key and new Error 'need fallback'
+    catch e
+      Lang.setLocale fallbackLocale
+      message = Lang.get key, replacements
+      Lang.setLocale currentLocale
     message
```
Comment (nanaya) [inline resources/assets/coffee/osu_common.coffee]: return in catch instead

File: resources/assets/coffee/osu_common.coffee [hunk 5]
```diff
@@ -216,11 +216,14 @@
...
-    message = Lang.get key, replacements, currentLocale
-
-    if message == key
-      message = Lang.get key, replacements, fallbackLocale
-
+    message = undefined
+    try
+      message = Lang.get key, replacements
+      message == key and new Error 'need fallback'
+    catch e
```
Comment (nanaya) [inline resources/assets/coffee/osu_common.coffee]: `e` not needed

**Comment by nanaya** [comment]
There is bigger problem (though probably not currently affected) and this doesn't quite fix it.

**Pull request merged**